tests/test_null_conversion.nim

```diff
@@ -0,0 +1,98 @@
+import os
+import macros
+import std/json
+import std/strutils
+import pkg/unittest2
+import stint
+
+import json_rpc/jsonmarshal
+
+import ../web3
+import ../web3/[conversions, ethtypes, engine_api_types]
+
+template should_be_value_error(input: string, value: untyped): void =
+  expect ValueError:
+    fromJson(%input, "", value)
+
+suite "Null conversion":
+  test "passing nully values to normal convertors":
+
+    ## Covers the converters which can be found in web3/conversions.nim
+    ## Ensure that when passing a nully value they respond with a ValueError
+
+    var resAddress: Address
+    var resDynamicBytes: DynamicBytes[32]
+    var resFixedBytes: FixedBytes[5]
+    var resQuantity: Quantity
+    var resRlpEncodedBytes: RlpEncodedBytes
+    var resTypedTransaction: TypedTransaction
+    var resUInt256: UInt256
+    var resUInt256Ref: ref UInt256
+
+      # Nully values
+    should_be_value_error("null", resAddress)
```

Comment:
As discussed in Discord, the most likely reason for the crashes in the object handling function in [nim-json-rpc/jsonmarshal.nim](https://github.com/status-im/nim-json-rpc/blob/master/json_rpc/jsonmarshal.nim#L43-L49), which is quite happy to propagate `nil` values to further calls to `fromJson`.

Your tests here are not covering it.

Reply:
Can you see the tests added below?